/**
 * Copyright 2019 Huawei Technologies Co., Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

// ms_manager.proto
syntax = "proto3";

package mindspore.serving.proto;
import "mindspore_serving/proto/ms_service.proto";

message AgentSpec {
  int64 rank_id = 1;
  int64 batch_size = 2;
  int64 input_size = 3;
  repeated Tensor inputs = 4;
  repeated Tensor outputs = 5;
}

message AgentRegisterRequest {
  repeated AgentSpec agent_spec = 1;
  string address = 2;
}

message AgentRegisterReply {
  ErrorMsg error_msg = 1;
}

message AgentExitRequest {
  repeated AgentSpec agent_spec = 1;
  string address = 2;
}

message AgentExitReply {
  ErrorMsg error_msg = 1;
}
